ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -59,24 +75,61 @@
     }
 
     final Number result = UPCASTER.get(targetType.baseType()).apply((Number) value);
-    return optional(result);
+    return Optional.of(result);
+  }
+
+  private static Optional<Object> coerceArray(final Object value, final SqlArray targetType) {
```

Comment:
```suggestion
  private static Optional<List<?>> coerceArray(final Object value, final SqlArray targetType) {
```

Or

```suggestion
  private static Optional<?> coerceArray(final Object value, final SqlArray targetType) {
```

Same for `coerceMap`